src/components/input/Select.tsx

```diff
@@ -311,31 +320,50 @@ function useListboxPositioning(
       buttonDistanceToBottom < listboxHeight &&
       buttonDistanceToTop > buttonDistanceToBottom;
 
-    if (asPopover) {
-      const { top: bodyTop } = document.body.getBoundingClientRect();
-      const absBodyTop = Math.abs(bodyTop);
-
-      return setListboxCSSProps({
-        minWidth: `${buttonWidth}px`,
-        top: shouldListboxDropUp
-          ? `calc(${absBodyTop + buttonDistanceToTop - listboxHeight}px - ${LISTBOX_TOGGLE_GAP})`
-          : `calc(${absBodyTop + buttonDistanceToTop + buttonHeight}px + ${LISTBOX_TOGGLE_GAP})`,
-        left:
-          right && listboxWidth > buttonWidth
-            ? `${buttonLeft - (listboxWidth - buttonWidth)}px`
-            : `${buttonLeft}px`,
-      });
+    if (!asPopover) {
+      // Set styles for non-popover mode
+      if (shouldListboxDropUp) {
+        return setListboxCSSProps({
+          bottom: '100%',
+          marginBottom: LISTBOX_TOGGLE_GAP,
+        });
+      }
+
+      return setListboxCSSProps({ top: '100%', marginTop: LISTBOX_TOGGLE_GAP });
     }
 
-    // Set styles for non-popover mode
-    if (shouldListboxDropUp) {
-      return setListboxCSSProps({
-        bottom: '100%',
-        marginBottom: LISTBOX_TOGGLE_GAP,
-      });
+    const { top: bodyTop, width: bodyWidth } =
+      document.body.getBoundingClientRect();
+    const absBodyTop = Math.abs(bodyTop);
+
+    // Calculate space available for the listbox in regular conditions, to make
+    // sure it never increases the body size, which could cause horizontal
+    // scrollbars to appear
+    const regularAvailableSpace =
+      (right ? buttonLeft + buttonWidth : bodyWidth - buttonLeft) -
+      LISTBOX_VIEWPORT_GAP;
+
+    let left = buttonLeft;
+    if (listboxWidth > regularAvailableSpace) {
+      // If the listbox is not going to fit the available space, let it "grow"
+      // in the opposite direction
+      left = right
+        ? LISTBOX_VIEWPORT_GAP
+        : left - (listboxWidth - regularAvailableSpace);
```

Comment:
```suggestion
        : left - listboxWidth - regularAvailableSpace;
```

Reply:
Actually, the parentheses are needed here, because they change the priority:

* `100 - (30 - 10)` -> `100 - 20` -> `80`
* `100 - 30 - 10` -> `70 - 10` -> `60`

![image](https://github.com/user-attachments/assets/f805f0c3-adb8-4452-b91f-fa1dc536cb22)

Reply:
🤦 indeed.